src/Features/Core/Portable/DocumentationComments/CodeFixes/AbstractAddDocCommentNodesCodeFixProvider.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.DiagnosticComments.CodeFixes
+{
+    internal abstract class AbstractAddDocCommentNodesCodeFixProvider
+        <TXmlElementSyntax, TXmlNameAttributeSyntax, TXmlTextSyntax, TMemberDeclarationSyntax> : CodeFixProvider
+        where TXmlElementSyntax : SyntaxNode
+        where TXmlNameAttributeSyntax : SyntaxNode
+        where TXmlTextSyntax : SyntaxNode
+        where TMemberDeclarationSyntax : SyntaxNode
+    {
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public async sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var parentMethod = root.FindNode(context.Span)?.FirstAncestorOrSelf<TMemberDeclarationSyntax>();
+
+            if (parentMethod != null)
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(
+                        c => AddParamTagAsync(context.Document, context.Span, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        protected abstract string NodeName { get; }
```

Comment:
They are available on a different range of types (rather than just `MemberDeclarationSyntax`), and they use a different syntax factory generator.  I thought it would be slightly messy to put all this in the one implementation of the abstract class.